test/junitTrace/JupiterTempDir.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfSystemProperty;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/*
+ * @test
+ * @bug 7903953
+ * @run junit JupiterTempDir
+ */
+class JupiterTempDir {
+    @Test
+    @EnabledIfSystemProperty(named = "test.name", matches = "JupiterTempDir.java")
+    void test(@TempDir Path temporary) throws Exception {
+        var currentWorkingDirectory = Path.of("");
+        var expected = currentWorkingDirectory.toAbsolutePath().toString();
+        var actual = temporary.getParent().toAbsolutePath().toString();
+        Assertions.assertEquals(expected, actual, "Unexpected root for temporary files");
```

Comment:
Would `Files.isSameFile(currentWorkingDirectory, temporary.getParent())` be more appropriate here?

Reply:
Not sure if this would hold true for every case, especially for some fancy (sym-)linked directories?

I think the `String`-based comparison is good-enough here, and has a nice side-effect producing a human-readable error messages when the assertion fails.